connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/recipients/RecipientsFilter.java

```diff
@@ -0,0 +1,62 @@
+package com.redhat.cloud.notifications.connector.email.processors.recipients;
+
+import com.redhat.cloud.notifications.connector.email.constants.ExchangeProperty;
+import com.redhat.cloud.notifications.connector.email.model.settings.RecipientSettings;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RecipientsFilter implements Processor {
+    /**
+     * Filters the fetched users according to the criteria specified in the
+     * recipient settings.
+     * @param exchange the exchange of the pipeline.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        // Fetch the required data to filter the users.
+        final RecipientSettings recipientSettings = exchange.getProperty(ExchangeProperty.CURRENT_RECIPIENT_SETTINGS, RecipientSettings.class);
+        final Set<String> subscribers = (Set<String>) exchange.getProperty(ExchangeProperty.SUBSCRIBERS, Set.class);
+        Set<String> usernames = exchange.getProperty(ExchangeProperty.USERNAMES, Set.class);
+
+        // If the request contains a list of usernames, then the recipients
+        // from RBAC who are not included in the request users list are
+        // filtered out. Otherwise, the full list of recipients from RBAC will
+        // be processed by the next step.
+        if (recipientSettings.getUsers().size() > 0) {
+            usernames = filterUsers(usernames, recipientSettings.getUsers());
+        }
+
+        // If the user preferences should be ignored, the recipients from RBAC
+        // (possibly filtered by the previous step) is returned without
+        // filtering out the users who didn't subscribe to the event type.
+        //
+        // TODO: Probably remove? Does it make sense?
+        if (recipientSettings.isIgnoreUserPreferences()) {
+            usernames = Set.copyOf(usernames);
+        }
+
+        exchange.setProperty(ExchangeProperty.USERNAMES, usernames);
+    }
+
+    /**
+     * Filters the users.
+     * @param users the set of users to filter.
+     * @param target the target users to filter from.
+     * @return the original set of users, but filtered, which only contains the
+     * users that are "allowed" in the specified target.
+     */
+    private Set<String> filterUsers(final Set<String> users, final Set<String> target) {
+        return users.stream()
+            .filter(
+                user -> target
+                    .stream()
+                    .anyMatch(requested -> requested.equalsIgnoreCase(user))
+            )
+            .collect(Collectors.toSet());
+    }
```

Comment:
Now that User is gone, this can be simplified:
```suggestion
    private void filterUsers(final Set<String> users, final Set<String> target) {
        users.retainAll(target);
    }
```

Reply:
Done!